### PR TITLE
Shorten generated blob identifiers

### PR DESCRIPTION
**Pull request by millerdev**
This should reduce the amount of RAM needed by the blob db backend. I have never once needed the filename that was previously embedded in the blob id, so I don't mind removing it.

Best to review commits individually. 51eb53d is refactoring only.

@snopoke @dannyroberts cc @proteusvacuum

### CODE REVIEW

File: corehq/blobs/tests/test_fsdb.py [hunk 1]
```diff
@@ -173,12 +137,60 @@ def test_empty_attachment_name(self):
...
+    @classmethod
+    def tearDownClass(cls):
+        cls.db = None
+        rmtree(cls.rootdir)
+        cls.rootdir = None
+
+    def test_put_with_colliding_blob_id(self):
+        # unfortunately can't do this on S3 because there is no way to
+        # reliably check if an object exists before putting it.
+        with patch("corehq.blobs.interface.random_url_id", new=lambda n:'not-unique'):
```
Comment (dimagimon) [inline corehq/blobs/tests/test_fsdb.py]: E231 missing whitespace after ':'

**Comment by snopoke** [comment]
I assume this will only apply to new attachments?

**Comment by millerdev** [comment]
Correct. We could write a migration to generate new ids for existing blobs if we wanted to save even more space.

File: corehq/blobs/interface.py [hunk 2]
```diff
@@ -86,13 +87,28 @@ def copy_blob(self, content, info, bucket):
...
+        change this if we ever expect an object to have significantly
+        more than 1000 blobs (attachments). The probability of a
+        collision with a 64 bit ID is:
+
+        k = 1000
+        N = 2 ** 64
+        (k ** 2) / (2 * 2 ** 64) = 2.7e-14
+
+        which is somewhere near the probability of a meteor landing on
+        your house. For most objects the number of blobs present at any
```
Comment (benrudolph) [inline corehq/blobs/interface.py]: 🌎 :comet:

**Pull request merged**